File: assignments/cash.c
# include <stdio.h>
# include <cs50.h>
# include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);
    int cents = round(dollars * 100);
    
    int coins = 0;
    
    while (cents != 0) 
    {
        // calculate quarters 1 quarter = 25c
        coins += cents / 25;
        cents = cents % 25;
        
        // calculate dimes 1 dime = 10c
        coins += cents / 10;
        cents = cents % 10;
        
        // calculate nickels 1 nickel = 5c
        coins += cents / 5;
        cents = cents % 5;
        
        // calculate pennies  1 penny = 1c
        coins += cents / 1;
        cents = cents % 1;
    }
    
    printf("%d\n", coins);
}

File: assignments/pset6/cash.py
from cs50 import get_float

change = 0.00

while change < .01:
    change = get_float("Change owed: ")

n = change * 100
count = 0

# Quarters
while n >= 25:
    count += 1
    n = n - 25

# Dimes
while n >= 10:
    count += 1
    n = n - 10

# Nickles
while n >= 5:
    count += 1
    n = n - 5

# Pennies
while n >= 1:
    count += 1
    n = n - 1

print(count)
